cvat/apps/engine/cache.py

```diff
@@ -64,10 +67,35 @@ def prepare_chunk_buff(self, db_data, quality, chunk_number):
                 chunk_number=chunk_number, chunk_size=db_data.chunk_size,
                 start=db_data.start_frame, stop=db_data.stop_frame,
                 step=db_data.get_frame_step())
-            for item in reader:
-                source_path = os.path.join(upload_dir, f"{item['name']}{item['extension']}")
-                images.append((source_path, source_path, None))
-
+            if db_data.storage == StorageChoice.CLOUD_STORAGE:
+                db_cloud_storage = db_data.cloud_storage
+                credentials = Credentials()
+                credentials.convert_from_db({
+                    'type': db_cloud_storage.credentials_type,
+                    'value': db_cloud_storage.credentials,
+                })
+                details = {
+                    'resource': db_cloud_storage.resource,
+                    'credentials': credentials,
+                    'specific_attributes': db_cloud_storage.get_specific_attributes()
+                }
+                cloud_storage_instance = get_cloud_storage_instance(cloud_provider=db_cloud_storage.provider_type, **details)
+                cloud_storage_instance.initialize_content()
+                for item in reader:
+                    name = f"{item['name']}{item['extension']}"
+                    source_path = os.path.join(upload_dir, name)
+                    if name not in cloud_storage_instance:
+                        raise Exception('{} file was not found on a {} storage'.format(name, cloud_storage_instance.name))
+                    cloud_storage_instance.download_file(name, source_path)
```

Comment:
@Marishka17 , Does it mean that our upload directory after some time will have the full copy of the cloud storage?